pkgs/development/tools/misc/cmake-language-server/default.nix

```
@@ -1,39 +1,33 @@
 { lib
 , buildPythonApplication
 , fetchFromGitHub
-, poetry-core
-, pythonRelaxDepsHook
 , cmake-format
 , pygls
 , cmake
+, pdm-pep517
 , pytest-datadir
 , pytestCheckHook
 }:
 
 buildPythonApplication rec {
   pname = "cmake-language-server";
-  version = "0.1.6";
+  version = "unstable-2023-01-08";
   format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "regen100";
     repo = pname;
-    rev = "refs/tags/v${version}";
-    sha256 = "sha256-B7dhCQo3g2E8+fzyl1RhaYQE6TFoqoLtp9Z7sZcv5xk=";
+    rev = "60c376a5fda29835060687569cb212350a292116";
+    sha256 = "sha256-vNG43sZy2wMetY5mbgxIoei5jCCj1f8vWiovWtwzbPc=";
   };
 
   patches = [
     # Test timeouts occasionally cause the build to fail
     ./disable-test-timeouts.patch
   ];
 
```

Hmm it looks like we can just add this to fix the cmake-language-server version:

```suggestion

  PDM_PEP517_SCM_VERSION = version;
```

but I think it only takes semver-style versions, this is what I get with the change:

```
> cmake-language-server --version
cmake-language-server 8
```

I wonder if we should just preemptively set this to 0.1.7?

---

Oh it actually uses [PEP 440](https://peps.python.org/pep-0440/), so I think we could then just set it as:

```suggestion

  PDM_PEP517_SCM_VERSION = "2013.1";
```

---

Added, thanks.